internal/cmd/handler.go

```diff
@@ -99,3 +116,63 @@ func PrintCMDResponseFull(request any, response core.APIResponse, responseType a
 		fmt.Println(util.MustToJSONIndent(responseContent))
 	}
 }
+
+// Check to see if a server is running and start a CMD server if it's not.
+// Returns (false, 0) if the primary server or CMD test server is already running,
+// (true, oldPort) if the CMD started its own server,
+// or log.Fatal() if another CMD server is already running.
+func mustEnsureServerIsRunning() (bool, int) {
+	statusInfo, err := common.CheckAndHandleServerStatusFile()
+	if err != nil {
+		log.Fatal("Failed to retrieve the current status file's json.", err)
+	}
+
+	if statusInfo != nil {
+		// Don't start the CMD server if the primary server or CMD test server is running.
+		if statusInfo.Initiator == common.PRIMARY_SERVER || statusInfo.Initiator == common.CMD_TEST_SERVER {
+			return false, 0
+		}
+
+		// log.Fatal() if another CMD server is running since they share the same working directory.
+		if statusInfo.Initiator == common.CMD_SERVER {
+			log.Fatal("Cannot start server, another CMD server is running.", log.NewAttr("PID", statusInfo.Pid))
+		}
```

Comment:
We will want to change up the structure here.
Note that it looks like there is a way for the code to get past both the conditionals without exiting the function, but there is not.
The core issue is that you have two mutually exclusive cases not in the same if tree.

I would use a switch here (switches are usually a go-to for enums).